### ISSUE TITLE
Add validation set for probing

### ISSUE THREAD

**Issue by ankeshanand** [mila-iqia/atari-representation-learning#14]
* We should have a validation set for probing to choose hyperparams.
* We should log validation metrics too for debugging. 
* An early stopping mechanism based on validation metrics would be very useful!
